### PR TITLE
Machine info introspects hugepages

### PR DESCRIPTION
**Pull request by derekwaynecarr**
I plan to add support for pre-allocated huge pages in Kubernetes 1.8.

In order to add that support, I need the ability to introspect the supported set of huge pages on the node, and the number of them that are pre-allocated.

This adds that support.

### CODE REVIEW

File: machine/info.go [hunk 1]
```diff
@@ -48,6 +50,43 @@ func getInfoFromFiles(filePaths string) string {
...
+			return hugePagesInfo, err
+		}
+
+		numFile := "/sys/kernel/mm/hugepages/" + st.Name() + "/nr_hugepages"
+		val, err := ioutil.ReadFile(numFile)
+		if err != nil {
+			return hugePagesInfo, err
+		}
+		var numPages uint64
+		n, err := fmt.Sscanf(string(val), "%d", &numPages)
```
Comment (sjenning) [inline machine/info.go]: any reason you use `fmt.Sscanf` here and `strconv.ParseUint` on 63?

File: machine/info.go [hunk 2]
```diff
@@ -48,6 +50,43 @@ func getInfoFromFiles(filePaths string) string {
...
+		if err != nil || n != 1 {
+			return hugePagesInfo, fmt.Errorf("could not parse file %v contents %q", numFile, string(val))
+		}
+		if err != nil {
```
Comment (sjenning) [inline machine/info.go]: this block can never be run; shorted by the previous block

File: machine/info.go (hunk 1 above)
Comment (derekwaynecarr) [inline machine/info.go]: ParseUint got tripped up by the \n in the file.

**Comment by dashpole** [comment]
rebase, and this should pass

**Comment by derekwaynecarr** [comment]
@dashpole -- thanks for updating the ci pieces... rebased, and it appears to pass.

File: machine/info.go [hunk 3]
```diff
@@ -48,6 +50,40 @@ func getInfoFromFiles(filePaths string) string {
...
+		}
+
+		numFile := "/sys/kernel/mm/hugepages/" + st.Name() + "/nr_hugepages"
+		val, err := ioutil.ReadFile(numFile)
+		if err != nil {
+			return hugePagesInfo, err
+		}
+		var numPages uint64
+		n, err := fmt.Sscanf(string(val), "%d", &numPages)
+		if err != nil || n != 1 {
```
Comment (dashpole) [inline machine/info.go]: can you add a comment explaining why we want to return error if n != 1?

File: machine/info.go [hunk 4]
```diff
@@ -48,6 +50,40 @@ func getInfoFromFiles(filePaths string) string {
...
+	}
+	for _, st := range files {
+		nameArray := strings.Split(st.Name(), "-")
+		pageSizeArray := strings.Split(nameArray[1], "kB")
+		pageSize, err := strconv.ParseUint(string(pageSizeArray[0]), 10, 64)
+		if err != nil {
+			return hugePagesInfo, err
+		}
+
+		numFile := "/sys/kernel/mm/hugepages/" + st.Name() + "/nr_hugepages"
```
Comment (dashpole) [inline machine/info.go]: nit: use a constant for the directory

**Comment by derekwaynecarr** [comment]
@dashpole - updated.  i would like us to return a value for each supported page size on the machine.  so in your case, a value of 0 for 2048kb at least lets me know your host supports 2Mi pages.

**Comment by dashpole** [comment]
thanks @derekwaynecarr.  I had misread the code: I thought it was checking the contents of the file, not the number of lines.
LGTM

**Pull request merged**